aws/ec2metadata/api.go

```diff
@@ -20,6 +20,20 @@ func (c *EC2Metadata) GetMetadata(p string) (string, error) {
 	return output.Content, req.Send()
 }
 
+// GetDynamicData uses the path provided to request
+func (c *EC2Metadata) GetDynamicData(p string) (string, error) {
+	op := &request.Operation{
+		Name:       "GetMetadata",
+		HTTPMethod: "GET",
+		HTTPPath:   path.Join("/", "dynamic", p),
+	}
+
+	output := &metadataOutput{}
+	req := c.NewRequest(op, nil, output)
+
+	return output.Content, req.Send()
```

Comment:
Or we could make `GetMetadata` call `http://169.254.169.254/` and leave the implementer to define `/meta-data`... but that would introduce breaking change, so I'm not sure that's a good idea.